flask_stupe/pagination.py

```diff
@@ -1,17 +1,34 @@
 import functools
 
 from flask import request
-
 from flask_stupe import pymongo
 
 __all__ = []
 
 
 if pymongo:
-    def _paginate(cursor, skip=None, limit=None, sort=None, count=True):
-        metadata = getattr(request, "metadata", None)
-        if count and isinstance(metadata, dict):
-            metadata.update(count=cursor.count())
+
+    def _get_link_header(total_count, limit, skip):
+        template = '<{}?limit={}&skip={{skip}}>; ' \
+                   'rel="{{rel}}"'.format(request.base_url, limit)
+        return ", ".join([
+            template.format(skip=0, rel="first"),
+            template.format(skip=skip - limit, rel="prev"),
+            template.format(skip=skip + limit, rel="next"),
+            template.format(skip=total_count - limit, rel="last")
+        ])
+
+    def _paginate(cursor, skip=None, limit=None, sort=None, count=True,
+                  headers=False):
+        if headers:
+            total_count = cursor.count()
+            headers = getattr(request, "response_headers")
+            headers["X-Total-Count"] = total_count
```

Comment:
Agreed & implemented for `X-Total-Count`.

I took a different approach for `Link`: it will be present in headers and populated only if `count`, `limit` & `skip` have been specified.

There is no added (since first pagination implementation) validation (yet?) on the values for `limit` & `skip`.